### ISSUE TITLE
[BUG] Tidal musicvideo download times out

### ISSUE THREAD

**Issue by steve1977** [nathom/streamrip#259]
### Describe the bug

Musicvideo download from tidal works for some videos, but others seem to time out. The log files asks me to post a bug report on this forum. Please see below:

https://pastebin.com/3CiWTYxR

### Command Used

```bash
https://pastebin.com/3CiWTYxR
```

### Debug Traceback

```text
https://pastebin.com/3CiWTYxR
```

### Config File

```toml
https://pastebin.com/3CiWTYxR
```

### Operating System

Linux

### streamrip version

1.0.15

**Comment by 89z** [comment]
please provide example URL

**Comment by steve1977** [comment]
Thanks for looking into this:

https://tidal.com/browse/video/205738663

**Comment by aberfan** [comment]
Likely related, but streamrip simply doesn't download music videos from Tidal at all.

I managed to trace through to get to where it's failing; it properly obtains the manifest file, which has the same urls as if you go through the website.  The issue seems to be that streamrip isn't passing proper (any?!) headers, and/or not properly rate-limiting. (at line 177 in downloadtools.py).   

Since I don't know python, I don't know how to correct it but the solution would be to properly send the cookies, etag, and other headers and to not pound the server¹ (unrelated to concurrency), since it looks like (from the ²403 Forbidden pages that it's getting instead of proper chunks) Tidal uses Varnish³ which is doing the dumping..

¹ https://stackoverflow.com/questions/35196974/aiohttp-set-maximum-number-of-requests-per-second
```xml
² <ClientResponse(https://vb-ad-fa.video.tidal.com/XXX/media_segment_1864384_6.ts?token=VALID_TOKEN) [403 Forbidden]>
<CIMultiDictProxy('Connection': 'close', 'Content-Length': '422', 'Server': 'Varnish', 'Retry-After': '0', '
Content-Type': 'text/html; charset=utf-8', 'Accept-Ranges': 'bytes', 'Date': 'Mon, 14 Feb 2022 12:52:45 GMT', 
'Via': '1.1 varnish', 'X-Served-By': 'cache-msp11867-MSP', 'X-Cache': 'MISS', 'X-Cache-Hits': '0', 
'X-Timer': 'S1644843166.896386,VS0,VE0')>
```
³ https://varnish-cache.org/intro/index.html#intro

**Comment by 89z** [comment]
I forgot about this thread for some reason. Regarding this video:

https://tidal.com/browse/video/205738663

that seems to just be a 30 second preview. Is that really what you're wanting? It doesn't seem worth the trouble to implement an extractor for that. Do you or anyone have a link to a full video of something?

**Comment by 89z** [comment]
Update, I kept getting old browser message from Tidal, so I tried changing my User-Agent to the newest Firefox or Chrome, but still got error. Then after a few more tries I got this message:

> You have been blocked.
> Why this blocking? Something about the behaviour of the browser has caught our attention.

So I cant get in now, with Web or Android client, even to access previous videos. So in short, fuck Tidal. I guess I will try again tomorrow if I remember. Good luck to everyone.
